Tasks/SonarQubePostTest/PRCA/PostComments-Server.ps1

```diff
@@ -327,4 +283,53 @@ function GetCodeFlowChangeTrackingId
     return $change.ChangeTrackingId
 } 
 
-#endregion 
\ No newline at end of file
+#endregion 
+
+#
+# Invoke a method with optional params by reflection. 
+# This can be used to overcome PS bugs in determining the correct overload candidate method from a .net assembly
+#
+function InvokeByReflection
+{
+    param ($obj, $methodName, [Type[]] $parameterTypes, [Object[]] $parameterValues)
+
+    $publicMethods = $obj.GetType().GetMethods() | Where-Object {($_.Name -eq $methodName) -and  ($_.IsPublic -eq $true)}
```

Comment:
A lot of the complexity and the reflection related code (and tests) can go away if we will leverage .net platform.